Roys/app/src/main/java/com/dwg_karrier/roys/Authentication.java

```diff
@@ -0,0 +1,245 @@
+  static final String OAUTHURL = "https://sandbox.feedly.com/v3/auth/auth";
+  static final String OAUTHSCOPE = "https://cloud.feedly.com/subscriptions";
+  static final String TOKENURL = "https://sandbox.feedly.com/v3/auth/token";
+  static final String GRANTTYPE = "authorization_code";
+  Context mainContext;
+
+  public Authentication(Context context) {
+    mainContext = context;
+  }
+
+  void authenticationAndBringPages() {
+    final Dialog auth_dialog;
+    WebView web;
+    auth_dialog = new Dialog(mainContext);
+    auth_dialog.setContentView(R.layout.auth_dialog);
+    web = (WebView) auth_dialog.findViewById(R.id.webv);
+    web.getSettings().setJavaScriptEnabled(true);
+    web.loadUrl(OAUTHURL + "?redirect_uri=" + REDIRECTURI + "&response_type=code&client_id=" + CLIENTID + "&scope=" + OAUTHSCOPE);
+    web.setWebViewClient(new WebViewClient() {
+      boolean authComplete = false;
+      String authCode;
+
+      @Override
+      public void onPageStarted(WebView view, String url, Bitmap favicon) {
+        super.onPageStarted(view, url, favicon);
+      }
+
+      @Override
+      public void onPageFinished(WebView view, String url) {
+        super.onPageFinished(view, url);
+        if (url.contains("?code=")) {
+          Uri uri = Uri.parse(url);
+          authCode = uri.getQueryParameter("code");
+
+          auth_dialog.dismiss();
+          new TokenGet(authCode, mainContext).execute();
+        } else if (url.contains("error=access_denied")) {
+          Log.i("", "ACCESS_DENIED_HERE");
+          auth_dialog.dismiss();
+        }
+      }
+    });
+    auth_dialog.show();
+  }
+
+  public JSONObject requestToken(String address, String token, String clientID, String clientSecret, String redirectUri, String grantType) {
+    JSONObject jObj = null;
+    try {
+      String params = "?code=" + token + "&client_id=" + clientID + "&client_secret=" + clientSecret + "&redirect_uri=" + redirectUri + "&grant_type=" + grantType;
+      URL url = new URL(address + params);
+      HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+      urlConnection.setRequestMethod("POST");
+      InputStream inputStream;
+
+      int status = urlConnection.getResponseCode();
+      if (status != HttpURLConnection.HTTP_OK)
+        inputStream = urlConnection.getErrorStream();
+      else
+        inputStream = new BufferedInputStream(urlConnection.getInputStream());
+
+      String result = convertStreamToString(inputStream);
+      try {
+        jObj = new JSONObject(result);
+      } catch (JSONException e) {
+        Log.e("JSON Parser", "Error parsing data " + e.toString());
+      }
+    } catch (MalformedURLException e) {
+      e.printStackTrace();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return jObj;
+  }
+
+  private class TokenGet extends AsyncTask<String, String, JSONObject> {
+    String code;
+    private ProgressDialog pDialog;
+    Context mainContext;
```

Comment:
I ask it because they don't have any scope specifier such as `public` or `private`. If that package-wide private scope is intended it's fine but otherwise I recommend you to specify it.